fe/fe-core/src/main/java/com/starrocks/qe/DDLStmtExecutor.java

```diff
@@ -335,9 +336,16 @@ public ShowResultSet visitCancelRefreshMaterializedViewStatement(CancelRefreshMa
 
         @Override
         public ShowResultSet visitAlterTableStatement(AlterTableStmt stmt, ConnectContext context) {
-            ErrorReport.wrapWithRuntimeException(() -> {
-                context.getGlobalStateMgr().alterTable(stmt);
-            });
+            String catalogName = stmt.getCatalogName();
+            if (CatalogMgr.isInternalCatalog(catalogName)) {
+                ErrorReport.wrapWithRuntimeException(() -> {
+                    context.getGlobalStateMgr().alterTable(stmt);
+                });
+            } else {
+                ErrorReport.wrapWithRuntimeException(() -> {
+                    context.getGlobalStateMgr().getMetadataMgr().alterTable(stmt);
```

Comment:
both of two catalogs can use `context.getGlobalStateMgr().getMetadataMgr().alterTable(stmt);`